google/resource_compute_address.go

```diff
@@ -119,22 +189,33 @@ func resourceComputeAddressRead(d *schema.ResourceData, meta interface{}) error
 }
 
 func resourceComputeAddressDelete(d *schema.ResourceData, meta interface{}) error {
+	computeApiVersion := getComputeApiVersion(d, AddressBaseApiVersion, AddressVersionedFeatures)
 	config := meta.(*Config)
 
 	addressId, err := parseComputeAddressId(d.Id(), config)
 	if err != nil {
 		return err
 	}
 
-	// Delete the address
-	log.Printf("[DEBUG] address delete request")
-	op, err := config.clientCompute.Addresses.Delete(
-		addressId.Project, addressId.Region, addressId.Name).Do()
-	if err != nil {
-		return fmt.Errorf("Error deleting address: %s", err)
+	var op interface{}
+	switch computeApiVersion {
+	case v1:
+		// Delete the address
+		log.Printf("[DEBUG] address delete request")
```

Comment:
Kind of weird that this log statement is only there for V1. Move it outside the switch case or add one in the beta code path